src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1007,8 +1010,10 @@ private unsafe void SetTcpKeepalive(SafeWinHttpHandle sessionHandle)
                     onoff = 1,
 
                     // Timeout.InfiniteTimeSpan will be converted to uint.MaxValue milliseconds (~ 50 days)
-                    keepaliveinterval = (uint)_tcpKeepAliveInterval.TotalMilliseconds,
-                    keepalivetime = (uint)_tcpKeepAliveTime.TotalMilliseconds
+                    // The result a of double->uint cast is unspecified and may differ on ARM, returning 0 instead of uint.MaxValue.
+                    // To handle Timeout.InfiniteTimespan correctly, we need to cast to Int32 first.
```

Comment:
```suggestion
                    // The result a of double->uint cast is unspecified for -1 and may differ on ARM, returning 0 instead of uint.MaxValue.
                    // To handle Timeout.InfiniteTimespan correctly, we need to cast to int first.
```